controllers/dnspolicy_dnsrecords.go

```diff
@@ -20,6 +20,43 @@ import (
 	"github.com/kuadrant/kuadrant-operator/pkg/multicluster"
 )
 
+type DNSMetricsEventHandler struct{}
+
+func (m *DNSMetricsEventHandler) OnCreate(_ context.Context, obj client.Object, err error) {
+	if err != nil {
+		return
+	}
+	dnsRecord, ok := obj.(*kuadrantdnsv1alpha1.DNSRecord)
+	if !ok {
+		return
+	}
+	dnsRecordEvent.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace, "create").Inc()
+}
+func (m *DNSMetricsEventHandler) OnUpdate(_ context.Context, obj client.Object, err error) {
+	if err != nil {
+		return
+	}
+	dnsRecord, ok := obj.(*kuadrantdnsv1alpha1.DNSRecord)
+	if !ok {
+		return
+	}
+	dnsRecordEvent.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace, "update").Inc()
+}
+func (m *DNSMetricsEventHandler) OnDelete(_ context.Context, obj client.Object, err error) {
+	if err != nil {
+		return
+	}
+	dnsRecord, ok := obj.(*kuadrantdnsv1alpha1.DNSRecord)
+	if !ok {
+		return
+	}
+	dnsRecordEvent.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace, "delete").Inc()
+}
+func (m *DNSMetricsEventHandler) OnStatusUpdate(_ context.Context, _ client.Object, _ error) {
```

Comment:
what are these for?

Reply:
just seemed like the right thing to do, if others want to add event handlers in the future, I don't currently need them for anything.